pynvim/msgpack_rpc/event_loop/asyncio.py

```diff
@@ -188,10 +188,42 @@ async def connect_stdout():
 
     @override
     def _connect_child(self, argv: List[str]) -> None:
+        def can_use_pidfd():
+            # Unix system without pidfd_open?
+            if not hasattr(os, 'pidfd_open'):
+                return False
+
+            # Check that we are not blocked by security policy like SECCOMP
+            try:
+                pid = os.getpid()
+                fd = os.pidfd_open(pid, 0)
+                os.close(fd)
+            except OSError:
+                return False
+
+            return True
+
+        def get_child_watcher():
+            if can_use_pidfd():
+                try:
+                    from asyncio.unix_events import PidfdChildWatcher
```

Comment:
```suggestion
                try:
                    from asyncio import PidfdChildWatcher, ThreadedChildWatcher
                    return PidfdChildWatcher()
                except ImportError:
                    # FIXME Python 3.7, return None if we drop support
                    return asyncio.get_child_watcher()
                     
```

On python3.14 (macos), `asyncio.unix_events.PidfdChildWatcher` symbol does not exist; there is only `_PidfdChildWatcher`.

Reply:
Are you sure that `can_use_pidfd()` returns `True` on MacOS?

Reply:
Actually no. Though symbols can be imported, I agree they won't work.

```python
>>> from asyncio.unix_events import can_use_pidfd
>>> can_use_pidfd()
False
>>> from asyncio import PidfdChildWatcher
ImportError: cannot import name 'PidfdChildWatcher' from 'asyncio'

>>> from asyncio.unix_events import _PidfdChildWatcher
>>> _PidfdChildWatcher()
<asyncio.unix_events._PidfdChildWatcher object at 0x105d11010>

>>> from asyncio.unix_events import _ThreadedChildWatcher
>>> _ThreadedChildWatcher()
<asyncio.unix_events._ThreadedChildWatcher object at 0x105d10c20>
```
and same for `ThreadedChildWatcher`. In the source code of python3.14 alpha distribution for macOS, those watcher classes are prefixed in `_`. So I think `asyncio.unix_events` is fine but platform-dependent. 

On my macOS python3.14 env, it falled back to `asyncio.get_child_watcher()` and error again (it's gone). It seems that our current fix works for Linux, but I'm not yet sure about macOS.

Reply:
Ah, I see. Pushed a fix.

Reply:
Thanks. I know it's not easy to reproduce from your end, but somehow  `asyncio.ThreadedChildWatcher` does not exist in python3.14. So it still falls back to `asyncio.get_child_watcher()` :/  I guess this is a regression in python3.14 dev versions --- because this works in python3.13 for me.

Since python3.14 is not a stable/released version, this shouldn't be a major blocker.